File: app/src/main/java/com/example/cbanda/achieve/Utilities/InjectorUtils.java
package com.example.cbanda.achieve.Utilities;

import android.content.Context;

import com.example.cbanda.achieve.Models.Repository.GoalRepository;
import com.example.cbanda.achieve.Models.Repository.GoalRepositoryImpl;
import com.example.cbanda.achieve.Models.db.GoalDatabase;

/**
 * Created by CBanda on 2018/02/08.
 */

public class InjectorUtils {

    private static GoalRepository goalRepository = null;

    public static GoalRepository getInstance(final Context context) {
        if (goalRepository == null) {
            synchronized (GoalRepository.class) {
                if (goalRepository == null) {
                    goalRepository = new GoalRepositoryImpl(GoalDatabase.getInstance(context).getGoalDao());
                }
            }
        }
        return goalRepository;
    }
}

File: app/src/main/java/com/example/cbanda/achieve/utilities/InjectorUtils.kt
package com.example.cbanda.achieve.utilities

import android.content.Context

import com.example.cbanda.achieve.models.db.GoalDatabase
import com.example.cbanda.achieve.models.repository.GoalRepository
import com.example.cbanda.achieve.models.repository.GoalRepositoryImpl
import com.example.cbanda.achieve.models.repository.QuoteRepository
import com.example.cbanda.achieve.models.repository.QuoteRepositoryImpl


object InjectorUtils {

    private var goalRepository: GoalRepository? = null
    private var quoteRepository: QuoteRepository? = null

    val instance: QuoteRepository
        get() {
            if (quoteRepository == null) {
                synchronized(GoalRepository::class.java) {
                    if (quoteRepository == null) {
                        quoteRepository = QuoteRepositoryImpl()
                    }
                }
            }
            return quoteRepository!!
        }

    fun getInstance(context: Context): GoalRepository {
        if (goalRepository == null) {
            synchronized(GoalRepository::class.java) {
                if (goalRepository == null) {
                    goalRepository = GoalRepositoryImpl(GoalDatabase.getInstance(context).goalDao)
                }
            }
        }
        return goalRepository!!
    }


}
